src/main/java/server/ServerTemp.java

```diff
@@ -23,18 +22,16 @@
     private final ServerHandler serverHandler;
     private final Codec codec;
     private final InetSocketAddress inetSocketAddress;
-    private final static Message welcomeMessage = new Message(MessageType.WELCOME, "[SERVER]", "Welcome to NioChat!");
-    private final static Message infoMessage = new Message(MessageType.WELCOME, "[SERVER]", "If you need some help just type /help");
     private final static Logger logger = LoggerFactory.getLogger(ServerTemp.class);
 
 
-    ServerTemp(int port, Codec codec, Viewer viewer) throws IOException {
+    ServerTemp(int port, Codec codec, Viewer viewer) throws Exception {
         this.port = port;
-        this.codec = codec;
+        this.codec = Objects.requireNonNull(codec, "Codec must not be null");
         inetSocketAddress = new InetSocketAddress(port);
         selector = Selector.open();
         serverSocketChannel = ServerSocketChannel.open();
-        serverHandler = new ServerHandler(selector, viewer);
+        serverHandler = new ServerHandler(viewer);
```

Comment:
The same as for the client. We probably need to pass the handler as a parameter in the constructor